submissions/7SergSmith7/Friends-app/app.js

```diff
@@ -0,0 +1,159 @@
+    <div class="contact__additional__gender">
+    <h4 class="additional__gender__title">sex</h4>
+    <p class="additional__gender_item">${gender}</p>
+    </div>
+    <div class="contact__additional__age">
+    <h4 class="additional__age__title">age</h4>
+    <p class="additional__age_item">${dob.age}</p>
+    </div>
+    <div class="contact__additional__location">
+    <h4 class="additional__location__title">location</h4>   
+    <p class="additional__location_item">${location.country}</p>     
+    </div>
+    
+    </div>
+</div>
+`;
+}
+
+function renderContacts(contacts) {
+  contactsField.innerHTML = contacts
+    .map((contact) => contactTemplate(contact))
+    .join("");
+}
+optionsMenu.addEventListener("click", onOptionsClick);
+
+function onOptionsClick(e) {
+  const selectedOption = e.target.classList.value;
+
+  switch (selectedOption) {
+    case "name-sort__z-a":
+      sortNameZa(tempContacts, filter);
+
+      break;
+    case "name-sort__a-z":
+      sortNameAz(tempContacts, filter);
+
+      break;
+    case "age-sort__more":
+      sortAgeMore(tempContacts, filter);
+
+      break;
+    case "age-sort__less":
+      sortAgeLess(tempContacts, filter);
+
+      break;
+    case "gender-filter__man":
+      filterMan(tempContacts);
+
+      break;
+    case "gender-filter__woman":
+      filterWoman(tempContacts);
+      break;
+    case "options__blocks__reset-btn":
+      resetBtn();
+      break;
+    case "options__search__icon":
+      searchName(searchInput);
+      break;
+    default:
+      break;
+  }
+}
+function resetBtn() {
+  copyData();
+  filter = "";
+  renderContacts(tempContacts);
+}
+
+function sortAgeMore(contact, filter) {
+  tempContacts = contact.sort((a, b) => a.dob.age - b.dob.age);
+  checkFilter(tempContacts, filter);
+}
+function sortAgeLess(contact, filter) {
+  tempContacts = contact.sort((a, b) => b.dob.age - a.dob.age);
+  checkFilter(tempContacts, filter);
+}
+function sortNameZa(contact, filter) {
+  tempContacts = contact.sort(function (a, b) {
+    if (a.name.first > b.name.first) {
+      return -1;
+    }
+    if (a.name.first < b.name.first) {
+      return 1;
+    }
+    return 0;
+  });
+  checkFilter(tempContacts, filter);
+}
+function sortNameAz(contact, filter) {
+  tempContacts = contact.sort(function (a, b) {
+    if (a.name.first < b.name.first) {
+      return -1;
+    }
+    if (a.name.first > b.name.first) {
+      return 1;
+    }
+    return 0;
+  });
+  checkFilter(tempContacts, filter);
+}
+
+function filterMan(contacts) {
+  renderContacts(contacts.filter((element) => element.gender == "male"));
+  filter = "man";
+}
+function filterWoman(contacts) {
+  renderContacts(contacts.filter((element) => element.gender == "female"));
+  filter = "woman";
+}
+function searchName(input) {
+  if (input.value)
+    renderContacts(
+      contactsData.filter((element) => element.name.first == input.value)
+    );
+  input.value = "";
+}
+
+function checkFilter(contacts, filter) {
```

Comment:
Why this function called `checkFilter`? Looks like it filter output by gender or renders contacts. So it is doing much more than just checking the filter. So naming is confusing.

Same with `filterWoman` and `filterMan` - this functions not only do filtering but also rendering contacts.

And this is a good example of spaghetti code :) In such a small application there is not so much issues with it, but in bigger code base you will quickly forget when you call `renderContacts` and when you just filter your data. To many responsibilities creates opaque functionality.

So what you need to do is:

1) Have 3 types of functions - data handlers, user actions handlers and render function
2) Data handlers, which is filter and sort functions should only filter or sort your global data
3) Render function should only render, like right now
4) User action handlers, event listeners, will be entry points to your logic and thier responsibility will be calling of other functions - they should call data handlers and render separately

In such case you will have plain logic, where all functions are working separately

 Good luck!

Reply:
I tried do with your advice.